localstack/services/cloudformation/provider.py

```diff
@@ -955,13 +989,25 @@ def delete_change_set(
         change_set_name: ChangeSetNameOrId,
         stack_name: StackNameOrId = None,
     ) -> DeleteChangeSetOutput:
+
+        # only relevant if change_set_name isn't an ARN
+        if not ARN_CHANGESET_REGEX.match(change_set_name):
+            if not stack_name:
+                raise ValidationError(
+                    "StackName must be specified if ChangeSetName is not specified as an ARN."
+                )
+
+            stack = find_stack(stack_name)
+            if not stack:
+                raise ValidationError(f"Stack [{stack_name}] does not exist")
+
         change_set = find_change_set(change_set_name, stack_name=stack_name)
         if not change_set:
-            return not_found_error(
-                f'Unable to find change set "{change_set_name}" for stack "{stack_name}"'
-            )
+            raise ChangeSetNotFoundException(f"ChangeSet [{change_set_name}] does not exist")
         change_set.stack.change_sets = [
-            cs for cs in change_set.stack.change_sets if cs.change_set_name != change_set_name
+            cs
+            for cs in change_set.stack.change_sets
+            if (cs.change_set_name != change_set_name and cs.change_set_id != change_set_name)
```

Comment:
tiny nit: could be simplified a bit
```suggestion
            if change_set_name not in (cs.change_set_name, cs.change_set_id)
```